FungibleToken.ts

```diff
@@ -127,10 +130,29 @@ export class FungibleToken extends TokenContract implements FungibleTokenLike {
     this.emitEvent("Transfer", new TransferEvent({ from, to, amount }))
   }
 
+  private permissionEquals(p1: Types.AuthRequired, p2: Types.AuthRequired) {
+    return p1.constant
+      .equals(p2.constant)
+      .and(p1.signatureNecessary.equals(p2.signatureNecessary))
+      .and(p1.signatureSufficient.equals(p2.signatureSufficient))
+  }
+
+  private checkPermissionsUpdate(update: AccountUpdate) {
+    let permissions = update.update.permissions
+
+    let { access, receive } = permissions.value
+    let accessIsNone = this.permissionEquals(access, Permissions.none())
+    let receiveIsNone = this.permissionEquals(receive, Permissions.none())
+    let updateAllowed = accessIsNone.and(receiveIsNone)
+
+    assert(updateAllowed.or(permissions.isSome.not()))
+  }
+
   @method
   async approveBase(updates: AccountUpdateForest): Promise<void> {
     this.paused.getAndRequireEquals().assertFalse()
     this.checkZeroBalanceChange(updates)
+    this.forEachUpdate(updates, (update, _usesToken) => this.checkPermissionsUpdate(update))
```

Comment:
I would do it like the wMina example and combine the logic into a single call to `this.forEachUpdate()`. Stepping through the tree of account updates comes with a lot of overhead (hashing of account updates!), about 10k constraints IIRC, so it's better to do it just once